psychopy/visual/textbox2/textbox2.py

```diff
@@ -1262,27 +1256,37 @@ def vertices(self):
             self.index = len(textbox._text)
         if self.index < 0:
             self.index = 0
-        # get the verts of character next to caret (chr is the next one so use
-        # left edge unless last index then use the right of prev chr)
-        # lastChar = [bottLeft, topLeft, **bottRight**, **topRight**]
+        # Get vertices of caret based on characters and index
         ii = self.index
         if textbox.vertices.shape[0] == 0:
-            verts = textbox._getStartingVertices() / textbox._pixelScaling
-            verts[:,1] = verts[:,1]
-            verts[:,0] = verts[:,0] + float(textbox._anchorOffsetX)
+            # If there are no chars, put caret at start position (determined by alignment)
+            if textbox.alignment[1] == "bottom":
+                bottom = min(textbox.contentBox._vertices.pix[:, 1])
```

Comment:
Do we need to do a min() here? Aren't all the chars on this row going to have the same vertical position?

Reply:
The vertices of `.contentBox` aren't character vertices, `.contentBox` is the inner rect (aka without padding), so its vertices are just the 4 corners of a rectangle. If it's reliable that the verts are in the same order I could just take one value, but I figured `min` would be safer, especially as there's only 4 values to search. Could also do `textbox.contentBox._pos.pix[1] - textbox.contentBox._size.pix[1]/2` but it seemed a bit wordy :P